Snowplow/Internal/Subject/SPPlatformContext.m

```diff
@@ -0,0 +1,121 @@
+#endif
+        [self setPlatformDict];
+    }
+    return self;
+}
+
+- (SPPayload *) fetchPlatformDict {
+#if SNOWPLOW_TARGET_IOS
+    @synchronized (self) {
+        NSTimeInterval now = [[NSDate date] timeIntervalSince1970];
+        if (now - _lastUpdatedEphemeralMobileDict >= _mobileDictUpdateFrequency) {
+            [self setEphemeralMobileDict];
+        }
+        if (now - _lastUpdatedEphemeralNetworkDict >= _networkDictUpdateFrequency) {
+            [self setEphemeralNetworkDict];
+        }
+    }
+#endif
+    return _platformDict;
+}
+
+- (long) getCountEphemeralMobileDictUpdates {
+    return _countEphemeralMobileDictUpdates;
+}
+
+- (long) getCountEphemeralNetworkDictUpdates {
+    return _countEphemeralNetworkDictUpdates;
+}
+
+// MARK: - Private methods
+
+- (void) setPlatformDict {
+    _platformDict = [[SPPayload alloc] init];
+    [_platformDict addValueToPayload:[SPUtilities getOSType]            forKey:kSPPlatformOsType];
+    [_platformDict addValueToPayload:[SPUtilities getOSVersion]         forKey:kSPPlatformOsVersion];
+    [_platformDict addValueToPayload:[SPUtilities getDeviceVendor]      forKey:kSPPlatformDeviceManu];
+    [_platformDict addValueToPayload:[SPUtilities getDeviceModel]       forKey:kSPPlatformDeviceModel];
+#if SNOWPLOW_TARGET_IOS
+    [self setMobileDict];
+#endif
+}
+
+- (void) setMobileDict {
+    [_platformDict addValueToPayload:[SPUtilities getCarrierName]         forKey:kSPMobileCarrier];
+    [_platformDict addValueToPayload:[SPUtilities getAppleIdfa]           forKey:kSPMobileAppleIdfa];
+    [_platformDict addValueToPayload:[SPUtilities getAppleIdfv]           forKey:kSPMobileAppleIdfv];
+    [_platformDict addNumericValueToPayload:[SPUtilities getTotalStorage] forKey:kSPMobileTotalStorage];
+    [self setEphemeralMobileDict];
+    [self setEphemeralNetworkDict];
+}
+
+- (void) setEphemeralMobileDict {
+    _lastUpdatedEphemeralMobileDict = [[NSDate date] timeIntervalSince1970];
+    _countEphemeralMobileDictUpdates++;
+    [_platformDict addNumericValueToPayload:[SPUtilities getBatteryLevel]       forKey:kSPMobileBatteryLevel];
+    [_platformDict addValueToPayload:[SPUtilities getBatteryState]              forKey:kSPMobileBatteryState];
+    [_platformDict addNumericValueToPayload:[SPUtilities isLowPowerModeEnabled] forKey:kSPMobileLowPowerMode];
+    [_platformDict addNumericValueToPayload:[SPUtilities getAvailableStorage]   forKey:kSPMobileAvailableStorage];
+    [_platformDict addNumericValueToPayload:[SPUtilities getAppAvailableMemory] forKey:kSPMobileAppAvailableMemory];
+}
+
+- (void) setEphemeralNetworkDict {
+    _lastUpdatedEphemeralNetworkDict = [[NSDate date] timeIntervalSince1970];
+    _countEphemeralNetworkDictUpdates++;
+    [_platformDict addValueToPayload:[SPUtilities getNetworkTechnology] forKey:kSPMobileNetworkTech];
+    [_platformDict addValueToPayload:[SPUtilities getNetworkType]       forKey:kSPMobileNetworkType];
```

Comment:
@AlexBenny I think it's a really good idea to clean up the SPUtilities and make SPPlatformContext self-sufficient. Just wanted to ask, should I deprecate the methods in SPUtilities once I move them to SPPlatformContext?

Reply:
Yes, I would deprecate them.
Usually, in the deprecated comment we can inform the developer of an alternative method to use but in this case it doesn't exist because the PlatformContext is private. However, I think that the method deprecation can be useful to warn the developers. If they raise an issue we may reconsider to expose them in an alternative way for v3.